src/Api/Parser/EventParsingIterator.php

```diff
@@ -69,36 +69,43 @@ private function parseEvent(array $event)
             if ($event['headers'][':message-type'] === 'error') {
                 return $this->parseError($event);
             }
+
             if ($event['headers'][':message-type'] !== 'event') {
                 throw new ParserException('Failed to parse unknown message type.');
             }
         }
 
-        if (empty($event['headers'][':event-type'])) {
+        $eventType = $event['headers'][':event-type'] ?? null;
```

Comment:
Are we guaranteed this will always be set?

Reply:
Actually my evaluation is wrong. We should even check for when this property is set but empty, which also should fail in that scenario.